### ISSUE TITLE
Support for CodeNarc Baseline feature

### ISSUE THREAD

**Issue by Artalus** [nvuillam/npm-groovy-lint#449]
I am planning to introduce `npm-groovy-lint` to a Jenkins shared library that has... quite a lot of issues, even when linted against `recommended-jenkinsfile` preset 😅 
I see that CodeNarc has some baseline support https://codenarc.org/codenarc-baseline-xml-report-writer.html , which should (to some extent) allow us to ignore the existing problems and focus on preventing new ones.

I first tried fiddling with the CodeNarc itself to generate the baseline:
```
$ docker run --rm -u "$(id -u):$(id -g)" -w=/w -v "$PWD":/w codenarc/codenarc -report=baseline:baseline.xml -rulesetfiles=file:/all.groovy
```
(`all.groovy` is just the downloaded https://codenarc.org/StarterRuleSet-AllRulesByCategory.groovy.txt)

This gave me a load of `Compilation failed for ...`, as CodeNarc was unable to resolve our Jenkins imports - but otherwise provided something that indeed looks like a baseline:
```xml
<?xml version="1.0" encoding="UTF-8"?><CodeNarc url="https://codenarc.org" version="3.6.0">
  <Report type="baseline"/>
  <Project title="">
    <SourceDirectory>.</SourceDirectory>
  </Project>
  <File path="vars/aws.groovy">
    <Violation ruleName="TrailingComma">
      <Message><![CDATA[List should contain trailing comma.]]></Message>
    </Violation>
    <Violation ruleName="SpaceAroundMapEntryColon">
      <Message><![CDATA[The colon for the literal Map entry for key [credentialsId] within class None is not followed by a match for regular expression [\S]]]></Message>
    </Violation>
...
  </File>
</CodeNarc>
```

Yet it seems I am unable to work with this thing from the `npm-groovy-lint` wrapper.
While `baseline` is a valid `-report` type for CodeNarc, `npm-groovy-lint`'s `--output` does not support it - and `-excludeBaseline=...` argument is not exposed to `npm-groovy-lint` CLI as well.
Seems that my only option is to be using `--codenarcargs`?  But it causes all of the `npm-groovy-lint` to be ignored, plunging me into the depths of raw CodeNarc madness instead of cozy wrapper CLI...

Do you think it would be possible to implement some experimental support for `--output=baseline:file.xml` and `--excludeBaseline=file.xml` on the linter's side?

**Comment by Artalus** [comment]
@nvuillam any comments on this one?

**Comment by nvuillam** [comment]
Sorry, free time is hard to find these weeks/months 🥹
